### DISCUSSION TITLE
Ask for feedback: User Interface Rework

### DISCUSSION THREAD

**Discussion by aronmolnar** [Syslifters/sysreptor#171] [Announcements]
🎉 The first draft of our UI rework is done.

Before we go to production, we kindly ask for your feedback.
Get a preview the UI at our playground: {no longer active}

Note that this is our testing branch. If you find any bugs, please just comment to this ticket.

We're looking forward to your feedback.

**Comment by vysecurity** [comment]
Don't like the dark mode tbh. Though UI looks good?

**Reply by aronmolnar** [reply]
In general or *this* dark mode?

You can switch to light mode via user account icon at top right

**Reply by vysecurity** [reply]
I think that the report will be in white background, black text. So it doesn't make sense to use a dark mode as it doesn't reflect the same feel.

**Reply by aronmolnar** [reply]
Got it.  
There's a light mode fortunately 😄

**Comment by gbrls** [comment]
I don't like that the other menu options are hidden this way. I liked that before it was easy to see basically all the menus and reach them within a single click. I don't like that in this new design the left bar hides those options

**Reply by aronmolnar** [reply]
I understand your point. We discussed at lot about this and how to:

* keep things as simple as possible
* still have enough space for all elements
* make it extendable (we're planning a plugin system where plugins might add new menu entries, which would not be possible in the top bar)

On the main page (project overview) the main menu entries are always visible.

![image](https://github.com/Syslifters/sysreptor/assets/25219407/05c06a30-95b2-4f33-a2f8-8547ae107a96)

In the reporting view, the main menu entries are now hidden, this is true.
You however reach all relevant main menu entries with one click.

![image](https://github.com/Syslifters/sysreptor/assets/25219407/4529c7f5-e859-4860-bfb4-10b89ce6ed9c)

The only menu entries you do not reach via one click (from a project) are Designs, Users and License information.

I agree that the advantage of having all main menu entries visible is that users always know what's still there (like a map of available functions).

Some things that we considered:
* Main menu entries in the top menu are not extendable
* Using this space for the report title gives us more space for reporting
* All important functions are reachable via one click
* Not so important functions can be hidden during report writing (but can be visible, if there is a good solution)
* In the reporting view, we have already two side menus (a third permanent side menu is not really an option)

We're happy to hear your ideas how to improve.

**Reply by vysecurity** [reply]
I also agree on the menu items somewhat. For some users you want the menu items to be static and accessible (there's not exactly a lot of menu items anyway). It's faster to directly click on the item than it is to open a menu and find it then click on it.

**Reply by aronmolnar** [reply]
This is true, but more menu items also come along with opportunity costs (more rarely used items take place and make it harder for users to find the often used elements).

Not sure how often you have to open those menu items (Designer, Users) from a project. (And if, you'll open it in a new tab.)

Technically, I think that it is not feasible to allow to make the main menu sticky, am I right @MWedl?

**Reply by MWedl** [reply]
What do you mean exactly with making the menu sticky? Adding a option to make the menu always visible (similar to the collapse/expand button in the second-level menus)? This would be technically feasible, but I'm not sure if this a good idea. When you are writing a report you then have 3 menus stacked on the left. These menus take up over 1/3rd of the screen width, leaving less space for actual content. The content area needs to be big in order to have enough space for writing markdown, especially in split-mode (with the MD-editor and preview side-by-side) or for smaller screens or when using two browser windows side-by-side (e.g. finding on the left side and notes to the right)). This proposal is contrary what we wanted to achieve with the UI rework: have more space for the main content area where you spend most time in, doing actual reporting tasks.

**Reply by aronmolnar** [reply]
We decided to wait for more feedback on this.

**Reply by gbrls** [reply]
Ok, now I understand why you did this. I'm not sure how to make it better, but this new way is very unintuitive for me, before it made much more sense, even from the first time that I opened the software. That's my feedback.

**Comment by JamHaven** [comment]
### Layout
The layout makes navigating smooth and I like the changes. Especially the folable menu on the left when working on a project. I would still love to see some improvements to the "+ Add"-Button at the bottom of the screen, especially on the notes page. Maybe some small "+" at the top ot the screen? Sometimes I have to look for that button, even though I use Sysreptor daily.

### Design:
I personally liked the blue look with the old logo more. The new design gives more of a "Another corporate software" look, which is fine, but I feel the heart is missing (little T-Rex I miss you already).

**Reply by aronmolnar** [reply]
Haha, thanks for your feedback.  

The little T-Rex is still in makeup and we plan to release its upgrade next year :)

Small pro-tip: You can add new notes using Ctrl+J

I wonder where to put an additional add button. In the personal notes it could be added here:
![image](https://github.com/Syslifters/sysreptor/assets/25219407/5cab8d63-6e86-4a47-b313-5fec151d5001)

...but disappear, if you collapse the side bar...

![image](https://github.com/Syslifters/sysreptor/assets/25219407/1122c62c-78ea-4c1c-8dec-de71f0d4a54c)

In the project notes it's more difficult, I think because we have no heading there...

![image](https://github.com/Syslifters/sysreptor/assets/25219407/cdc8b667-ba99-4cf0-a634-2c8cf569ece7)

I would find a button at the top unintuitive there and somewhat redundant.  
Not sure 🤔

**Reply by JamHaven** [reply]
Can't wait for the upgraded T-Rex ;)

Good tip with the shortcut, thx!

I know, fitting another button somewhere at the top ist quite the challenge...
Here is a screenshot of my view during notekeeping/reporting to add a little bit of context why I think the Button should move:
![image](https://github.com/Syslifters/sysreptor/assets/17318066/7f505415-b6be-4603-8277-e398945592e8)

A "+ Add" Button at the top feels better to navigate in my own opinion, but It takes away from the actual content.
![image](https://github.com/Syslifters/sysreptor/assets/17318066/b26066a4-7e74-4ee8-a799-1256ed9669a4)

This could also be an option, but i personally don't like it:
![image](https://github.com/Syslifters/sysreptor/assets/17318066/35623d86-8323-4231-abfb-3a29903402ee)

**Reply by aronmolnar** [reply]
The first option works, but then I would remove the lower Add button. (The second does not work. Nobody would find this button and expect something else.)

Notion has two buttons (at the top and the bottom, but due to the different wording I expect different things; not solved perfectly, I believe).

![image](https://github.com/Syslifters/sysreptor/assets/25219407/b0614479-63fe-44e7-8781-9f7a446a3dfe)

I like the solution Joplin provides. At the top a search bar and different buttons for checks and for notes.  
The downside is, that it's not clearly communicated that those actions add new notes.

![image](https://github.com/Syslifters/sysreptor/assets/25219407/b151711a-d683-4377-b462-293a561af6de)

This might be improved by better icons, e.g.
![image](https://github.com/Syslifters/sysreptor/assets/25219407/19a084a5-b9d1-4b2d-95ae-74b28a5b5695)

![image](https://github.com/Syslifters/sysreptor/assets/25219407/85103aa5-2a95-49ee-a9c5-6292698a19fe)

![image](https://github.com/Syslifters/sysreptor/assets/25219407/c8da79e0-37e4-45ca-879b-3d43e499c2fe)


What I like about this is that it probably improves the adoption of checklists.

Obsidian follows a similar approach.

![image](https://github.com/Syslifters/sysreptor/assets/25219407/bac29d39-9ddf-4d2e-ad01-e5433aed6cfe)


What do you think, @MWedl?

**Reply by cmahrl** [reply]
I like the idea of an action bar at the top. Icons are often self-explanatory. We also considered this in a UI draft. However, it should be consistent throughout the UI e.g. this should then also be considered for the reporting sidebar and the notes sidebar in a project. Because currently, if you want to add a finding, you would also find the Add button at the bottom (same case).

**Reply by MWedl** [reply]
The add-button on top will work fine for notes. For findings, however, it does not make sense to move the add-button to the top, because there are also sections. If we moved the button to the top, the order would be: 1. add finding button, 2. section list, 3. finding list. 

We could add the button as a small icon besides the findings sub-headline in the list (similar to the change sort-mode button). I do not like this solution, because the add-button would be much smaller, less prominent and harder to click.

**Reply by aronmolnar** [reply]
I also believe that there cannot be a fully consistent solution for notes and findings.

I think, however, that an additional button at the findings section could make sense. If we add it to the right of the "Findings" title (instead of beside the sort icon), we could make it a bit larger. But any solution there must be supplementary.

![image](https://github.com/Syslifters/sysreptor/assets/25219407/19160b9a-75eb-418f-bafe-2e331385e03c)

**Reply by aronmolnar** [reply]
For notes we will remove the button at the bottom and create two icons to create new notes at the top.

For findings, we will add a "+" icon right beside the "Findings" heading but we'll additionally have the existing "Add" button at the bottom.

We take the inconsistency into account.

**Comment by cmahrl** [comment]
it could also be a moving Add button, that sticks to the button, when to many finding / notes are added. Could work for both Notes and Findings. So at the beginning, the button would be at the top. 

IMHO I'd go for something consistent. Having to search the add button on different sub pages with very similar behavior is not ideal.

![image](https://github.com/Syslifters/sysreptor/assets/7896159/fb83e1a0-3842-4a79-9416-b831f1c03f6f)

**Comment by aronmolnar** [comment]
User interface rework was deployed with last update.
